src/Scaler/Services/ScalerMetadata.cs

```diff
@@ -4,17 +4,33 @@
 
 namespace Keda.CosmosDb.Scaler
 {
-    [JsonObject(ItemRequired = Required.Always)]
+    [JsonObject]
     internal sealed class ScalerMetadata
     {
-        private string _metricName;
-
-        public string Connection { get; set; }
+        private string _metricName; // Private backing field for MetricName property
+        private string _connection; // Private backing field for Connection property
+        public string ConnectionFromEnv { get; set; }
+        public string Connection
+        {
+            get => string.IsNullOrEmpty(_connection) ? ConnectionFromEnv : _connection;
```

Comment:
Should be okay to treat empty (`""`) connection string just like any invalid connection string as it will rarely happen that someone inputs empty value. Can implement the getter as `get => _connection ?? this.ConnectionFromEnv;`.